start

```diff
@@ -60,18 +62,78 @@ cat >> ${PG_CONFDIR}/pg_hba.conf <<EOF
 host    all             all             0.0.0.0/0               md5
 EOF
 
+# allow replication connections to the database
+if [ -n "${REPLICATION_USER}" ]; then
+  if [ "${PSQL_SSLMODE}" == "disable" ]; then
+    cat >> ${PG_CONFDIR}/pg_hba.conf <<EOF
+host    replication     $REPLICATION_USER       0.0.0.0/0               md5
+EOF
+  else
+    cat >> ${PG_CONFDIR}/pg_hba.conf <<EOF
+hostssl replication     $REPLICATION_USER       0.0.0.0/0               md5
+EOF
+  fi
+fi
+
+if [ "${PSQL_MODE}" == "master" ]; then
+  if [ -n "${REPLICATION_USER}" ]; then
+    echo "Supporting hot standby..."
+    cat >> ${PG_CONFDIR}/postgresql.conf <<EOF
+wal_level = hot_standby
+max_wal_senders = 3
+checkpoint_segments = 8
+wal_keep_segments = 8
+EOF
+    if [ "${PSQL_ARCHIVEMODE}" == "on" ]; then
```

Comment:
@Downchuck I believe the `PSQL_ARCHIVEMODE` change is not related to the replication feature right? I suggest this be submitted as a separate PR for reasons mentioned earlier.

However I tried adding `-e 'PSQL_ARCHIVEMODE=on'` in the launch command of the master and got these errors.

```bash
Enabling trust samenet in pg_hba.conf...
Supporting hot standby...
Creating user "replicator"...
LOG:  unrecognized configuration parameter "archive_cleanup_command" in file "/etc/postgresql/9.4/main/postgresql.conf" line 622
FATAL:  configuration file "/etc/postgresql/9.4/main/postgresql.conf" contains errors
```